src/oge/data_cleaning.py

```diff
@@ -2145,3 +2146,349 @@ def aggregate_cems_to_subplant(cems):
+    """Screen timeseries for anomalous value following the algorithm steps described
+    in Tyler H. Ruggles et al. Developing reliable hourly electricity demand data
+    through screening and imputation (2020)
+
+    Note that the screening algorithms have been developed for demand timeseries and
+    some of these algorithms might not be tailored to other time series.
+
+    The screening is conducted in 2 steps. Step 1 removes the most egregious anomalies
+    where few or no calculations are needed. Afterward, in Step 2, the most extreme
+    values have been removed making calculations of local characteristics of the data
+    more reasonable. Through this screening process hourly values can be recategorized
+    from okay to other classifications based on the algorithms.
+    """
+
+    def __init__(
+        self, df: pd.DataFrame, field: str, **kwargs: Optional[dict[str, Any]]
+    ):
+        """Constructor
+
+        Arguments
+        ---------
+        * `df` : timeseries as a data frame.
+        * `field` : Column to screen.
+        * `kwargs` : Optional keyword arguments.
+        """
+        self.df = df.copy().reset_index()
+        self.field = field
+        self.global_cut_multiplier = (
+            kwargs["global_cut_multiplier"] if "global_cut_multiplier" in kwargs else 10
+        )
+
+        self.add_category()
+
+    def add_category(self):
+        """Add category label to track which algorithm screens an hourly value"""
+        self.df["category"] = np.where(self.df[self.field].isna(), "MISSING", "OKAY")
+        self.df = self.df.assign(missing=self.df[self.field].isna())
+
+    def flag_negative_values(self):
+        """Flag negative values"""
+        flag = self.df[self.field] < 0
+        self.df["category"] = np.where(flag, "NEGATIVE", self.df["category"])
+        filtered = np.where(flag, self.df[self.field], np.nan)
+
+        self.df["negative_filtered"] = filtered
+        self.df[self.field] = self.df[self.field].mask(flag)
+
+    def flag_zero_values(self):
+        """Flag zero values"""
+        flag = self.df[self.field] == 0
+        self.df["category"] = np.where(flag, "ZERO", self.df["category"])
+        filtered = np.where(flag, self.df[self.field], np.nan)
+
+        self.df["zero_filtered"] = filtered
+        self.df[self.field] = self.df[self.field].mask(flag)
+
+    def flag_runs(self):
+        """flag the third hour onwards as anomalous in a series of identical values"""
+
+        d1 = self.df[self.field].diff(periods=1)
+        d2 = self.df[self.field].diff(periods=2)
+
+        filtered = self.df[self.field].mask((d1 == 0) & (d2 == 0))
+        self.df["run_filtered"] = np.where(
+            self.df[self.field] != filtered, self.df[self.field], np.nan
+        )
+        self.df[self.field] = filtered
+        self.df["category"] = np.where(
+            self.df["run_filtered"].notna(), "IDENTICAL_RUN", self.df["category"]
+        )
+
+    def flag_global_extreme_values(self):
```

Comment:
For global extremes for generation data, we may not want to use the median of all values. We may want to only take the median of all non-zero values. I say this because many plants may sit idle for large parts of the year, and so the median value might actually be zero, and we would not want to flag any non-zero value

Reply:
Ok. I will add a boolean argument that will exclude or not zero-values.

Reply:
In fact, I realize that this is already implemented via the `flag_zero_values` method.